middleware/userValidationRegex.js

```diff
@@ -0,0 +1,26 @@
+const regex = (req, res, next) => {
+  try {
+    const name = /^[a-zA-Z ]{2,30}$/;
+    const gender = /^male$|^female$|^other$/;
+    const dob = /^([0-2][0-9]|(3)[0-1])(\/)(((0)[0-9])|((1)[0-2]))(\/)\d{4}$/;
+    const userName = /^[a-zA-Z ]{2,30}$/;
+    const email = /^\w+([\\.-]?\w+)*@\w+([\\.-]?\w+)*(\.\w{2,3})+$/;
+    const password =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})/;
+    if (
+      name.test(req.body.name) &&
```

Comment:
Test one param at a time and send a suitable message for each param.

Reply:
Checking indivisual condition and giving corrusponding response